### PR TITLE
skip re-building existing forecast outputs

### LINKED ISSUE #32: Add catch to `generate_forecast_json_file()` to only generate if the file does not exist
Right now, the builds for the FluSight forecast hub take roughly 9 minutes to complete, but much of this time is spent re-building forecast rounds that have already been built.

Instead, we could check and skip the JSON generation if the file exists and is not the current round. This would drastically reduce the amount of time it takes to run the script for an existing dashboard.

### PR DESCRIPTION
**Pull request by zkamvar**
I tested this out and found that this provides a speedup if we skip rebuilding the JSON files. As in building the FluSight hub data takes 3 minutes instead of 9 minutes.

For example [this run processed the most recent target data in 3 minutes](https://github.com/hubverse-org/hub-dashboard-control-room/actions/runs/12017083557/job/33498669950) using the code from this PR.

This will fix #32

### CODE REVIEW

**Comment by zkamvar** [comment]
I'll come back to this on Tuesday.

**Comment by matthewcornell** [comment]
@zkamvar , @elray1  and I spoke after his comment, and we agreed that the --rebuild arg could wait until we merge this current work. This would allow me to move forward with getting the app working with https://github.com/CDCgov/covid19-forecast-hub . That said, I'm happy to pair on adding that feature now. The click code would look something like this, I believe:
```python
import click

@click.command()
@click.option('--verbose', is_flag=True, help='Enable verbose output.')
def hello(verbose):
    if verbose:
        click.echo('Verbose mode enabled!')
    click.echo('Hello, world!')

if __name__ == '__main__':
    hello()
```

**Comment by matthewcornell** [comment]
I'd also suggest 1) adding parens (I understand that making logic explicit like this is a code style preference, which as you point out is not captured in the lab anywhere), and 2) reversing test inequality order. I.e., change to:

```python
    if (reference_date != current_date) and Path(json_file_path).exists():
```

Or if we add the above rename in:
```python
    if (reference_date != newest_reference_date) and Path(json_file_path).exists():
```

**Comment by matthewcornell** [comment]
My last comment is that I'm confused by the `current_date != reference_date` part of this logic (though @elray1 was OK with it):
```python
    if current_date != reference_date and Path(json_file_path).exists():
```

I expected that the `exists()` would be enough to check. Doesn't the inequality assume that the only change is adding data for only the newest reference date? I guess this goes to the scenario in which a modeler "back updates" a forecast. Anyway, no action - just looking for some intuition around the logic.

**Comment by matthewcornell** [comment]
Thanks for the test! I'm thinking we could omit the `assert set(json_files) == ...` block because it seems to be essentially testing that the unlink/delete worked. ?

**Review by matthewcornell — APPROVED** [review]

**Comment by zkamvar** [comment]
Thank you for working with me on this @matthewcornell!

**Pull request merged**
